hack/RELEASE.md

```diff
@@ -0,0 +1,23 @@
+# Libpod/Podman/Podman Remote Release Checklist
+
+0. Make a new branch using the name format <1.2.3>[-<epic>]
+1. Open a PR with a commit having a message in the following format:
```

Comment:
Release process is two PRs: a release notes PR, and a release PR.

Release notes updates RELEASE_NOTES.md and adds release notes, as one might expect. This must merge before the second PR is created.

We want two commits in the release PR, which must be rebased on top of the latest version of the branch (IE, include the release notes PR that merged before):

First one modifies version/version.go and contrib/spec/podman.spec.in and changes the version numbers there to the desired release, plus runs `make changelog` to generate changelog. Second one modifies version/version.go and the RPM spec again, bumping the micro version and appending `-dev` to version.go.

The first commit should be PGP signed, be named v2.0.1 or similar (the version number with lowercase v prepended), and have a message the same as the name of the tag. This tag is manually pushed once the second PR merges.

Reply:
For example, the tag command I used to generate the v2.0.1 tag:

`git tag -s -m 'v2.0.1' v2.0.1 <FULL_COMMIT_SHA>`

Reply:
Can all the commits come in under a single PR?  From a process-triggering perspective, that's much more simple to detect.

Rebasing the code and creating tags/branches/github-releases are all fairly well-understood puzzle-pieces for me.  I also intend to include release-validation checks in the workflow that occurs in the PR.  So you'll have quick feedback before anything goes upstream/official.

I also want to wire-in notifications at the master and release-branch level, so we know right away if something breaks.  Github-actions provides a 'retry' button just like cirrus does.  But having confidence in the automation at the PR-stage is an important goal of mine.

Reply:
The release notes shouldn't require any action from release automation, but if we have a process doc, they should be mentioned.

All action should be on the actual release PR, which should always be a consistent format.

Reply:
Excellent, glad to hear it. That also means release-commit / PR problems can be caught by the dummy-release run inside the PR.  Or the dummy run will be missing...which will also be noticed.  Okay, I believe I know the way forward here.